README
MegaPoke - phpFox Poke Module by phpfoxLABS.com

MegaPoke CURRENT Features:
- Unlimited Poke Types (added 04/15/11)
- Feed and Notification Support (added 04/15/11)
- Email Notification (added 04/15/11)
- All Phrases in Phrase Manager (added 04/15/11)
- User Group settings for "Posting to Feed" and to "Disable MegaPoke" (added 04/15/11)
- User Privacy Settings to turn off MegaPoking on profile. (added 04/15/11)
- User Notification Settings to stop getting Poke Notification Emails (added 04/15/11)
- Flood Control to stop over MegaPoking (added 04/15/11)
- Ryan's Advanced Feed Support (added 04/22/11)

MegaPoke FUTURE Features:
- Add Images along with Poke Types
- Replace View in Notification with "Poke Back"

No Core Modifications: Just upload, install in Admin CP, add Poke Types, and start poking!

Demo "MegaPoke" on our Demo Site:
http://phpfoxlabs.com/v2/ 

Checkout our other phpFox Modules:
http://phpfoxlabs.com/module.list/ 

Need custom phpFox development?  Feel free to contact us about your project:
Phone: +1 (407) 490-3223
Email: sales@phpfoxlabs.com
Chat Live: visit http://phpfoxlabs.com and click Live Chat

#######################################################################

This work is licensed under a Creative Commons Attribution-ShareAlike 3.0 Unported License.
human-readable summary: http://creativecommons.org/licenses/by-sa/3.0/
full legal code: http://creativecommons.org/licenses/by-sa/3.0/legalcode

#######################################################################


HISTORY
MODULE UPDATE HISTORY:

4/22/11 - ADDED: Ryan's Advanced Feed Support

4/17/11 - BUG: DB table not being created for some users.   FIX: removed out of place "/" at the end of install script.  Was NOT inside DB SQL creation and no harm could of happened to users DB. - Reported by: jackparr52 and finally
https://github.com/phpfoxLABS/MegaPoke/commit/81598abd2bf4543d9c6b12ff82df7e8fd62b23e8#upload/include/xml/phpfox_megapoke.xml

4/16/11 - Turned off "Can Drag/Drop" on MegaPoke Profile Block

4/15/11 - CORE MODULE ADDED




upload/module/megapoke/include/service/callback.class.php
<?php
 /**
 * 
 * @copyright		phpfoxLABS.com
 * @author  		Jacob Adams
 * @package  		MEGAPOKE
 * @version 		v1.0.0 - 2011-04-14
 *
 * This work is licensed under a Creative Commons Attribution-ShareAlike 3.0 Unported License.
 * human-readable summary: http://creativecommons.org/licenses/by-sa/3.0/
 * full legal code: http://creativecommons.org/licenses/by-sa/3.0/legalcode
 *
 *
 */


defined('PHPFOX') or exit('NO DICE!');

class Megapoke_Service_Callback extends Phpfox_Service 
{
	
	public function __construct()
	{	
		$this->_sTable = Phpfox::getT('megapoke');
	}
	
		
	public function getProfileSettings()
	{
		return array(
			'megapoke.who_can_poke' => array(
				'phrase' => Phpfox::getPhrase('megapoke.settings_who_can_poke'),
				'default' => '0'				
			)
		);
	}
		
		
		
	public function getNotificationSettings()
	{
		return array('megapoke.new_poke_email' => array(
				'phrase' => Phpfox::getPhrase('megapoke.new_poke_notification_email'),
				'default' => 1
			)
		);		
	}
	
	
	
		
	public function getNotificationFeed($aRow)
	{	
	
	
		$oUrl = Phpfox::getLib('url');
		$oParseOutput = Phpfox::getLib('parse.output');	
	
		$aPoke =phpfox::getLib('phpfox.database')->select('*')
		->from(Phpfox::getT('megapoke'))
		->where("poke_type_id = '".$aRow['item_id']."' ")
		->execute('getSlaveRow');
		
		$aUserFrom = Phpfox::getLib('phpfox.database')->select(str_replace("u.", "", Phpfox::getUserField()))
		 ->from(Phpfox::getT('user'))
		 ->where("user_id = '". Phpfox::getUserId()."'")
		 ->execute('getSlaveRow');
		 
		 
		
		$aRow['link'] = $oUrl->makeUrl('feed.user', array('id' => $aRow['user_id']));
		$aRow['message'] = Phpfox::getPhrase('megapoke.notification', array(
				'user_link' => $oUrl->makeUrl('feed.user', array('id' => $aRow['user_id'])),
				'owner_full_name' => Phpfox::getLib('parse.output')->clean($aRow['full_name']),
				'content' => $aPoke['poke_text']
			)
		);	
	
	
	
		
		return $aRow;
	
	}	
		
		
	public function getNewsFeed($aRow)
	{	
	
	
		$oUrl = Phpfox::getLib('url');
		$oParseOutput = Phpfox::getLib('parse.output');	
	
	
		$aRow['text'] = Phpfox::getPhrase('megapoke.ijustsentapoke', array(
				'user_link' => $oUrl->makeUrl('feed.user', array('id' => $aRow['user_id'])),
				'owner_full_name' => Phpfox::getLib('parse.output')->clean($aRow['owner_full_name']),
				'viewer_link' => $oUrl->makeUrl('feed.user', array('id' => $aRow['viewer_user_id'])),
				'viewer_full_name' => Phpfox::getLib('parse.output')->clean($aRow['viewer_full_name']),
				'content' => $aRow['content']
			)
		);	
	
		$aRow['enable_like'] = true;
		
		return $aRow;
	
	}
	
	
}

?>

upload/module/megapoke/template/default/block/profilepoke.html.php
<?php 
 /**
 * 
 * @copyright		phpfoxLABS.com
 * @author  		Jacob Adams
 * @package  		MEGAPOKE
 * @version 		v1.0.0 - 2011-04-14
 *
 * This work is licensed under a Creative Commons Attribution-ShareAlike 3.0 Unported License.
 * human-readable summary: http://creativecommons.org/licenses/by-sa/3.0/
 * full legal code: http://creativecommons.org/licenses/by-sa/3.0/legalcode
 *
 */
defined('PHPFOX') or exit('NO DICE!'); 

?>
{if !count($aPokes)}

{else}
<form action="{$pokeFormUrl}" name="pokeme" method="post">
<input type="hidden" name="poke_user" value="{$aUser.user_name}" />
<input type="hidden" name="poke_userid" value="{$aUser.user_id}" />
<input type="hidden" name="poke_user_full_name" value="{$aUser.full_name}" />
<table border="0" cellspacing="2" cellpadding="2">
  <tr>
    <td>
    	{$sSendText}</td>
    <td>
    <select onchange="this.form.submit();" name="poke_type_id">
    <option>----</option>
    {foreach from=$aPokes item=v} 
   		 <option value="{$v.poke_type_id}">{$v.poke_text}</option>
    {/foreach}
    </select>
    
    
    </td>
  </tr>
</table>
</form>
{/if}

upload/module/megapoke/include/component/block/profilepoke.class.php
<?php

 /**
 * 
 * @copyright		phpfoxLABS.com
 * @author  		Jacob Adams
 * @package  		MEGAPOKE
 * @version 		v1.0.0 - 2011-04-14
 * 
 * This work is licensed under a Creative Commons Attribution-ShareAlike 3.0 Unported License.
 * human-readable summary: http://creativecommons.org/licenses/by-sa/3.0/
 * full legal code: http://creativecommons.org/licenses/by-sa/3.0/legalcode
 * 
 *
 */


defined('PHPFOX') or exit('NO DICE!');


class Megapoke_Component_Block_profilepoke extends Phpfox_Component
{
	
	public function process()
	{
		$aUser = $this->getParam('aUser');
		
		#
		# Check User Privacy Settings
		if (!Phpfox::getService('user.privacy')->hasAccess($aUser['user_id'], 'megapoke.who_can_poke'))
		{
			return false;
		}		
		
		
		
		# Check for User Group Permission
		if (!Phpfox::getUserParam('megapoke.canmegapoke')){ return false; }	 
		
		# User can't Poke themself
		if ($aUser['user_id'] == Phpfox::getUserId()){ return false; } 
		
		# User must be logged in to Poke
		if (!Phpfox::getUserId()){	return false; } 
		
	
	
	
		#GET POKES
		$aPokes =phpfox::getLib('phpfox.database')->select('*')
		->from(Phpfox::getT('megapoke'))
		->where("poke_active = '0' ")
		->order('poke_text')
		->execute('getSlaveRows');	
	
	
		$this->template()->assign(array(
				'sHeader' => Phpfox::getPhrase('megapoke.profilepoketitle'),
				'aPokes' => $aPokes,
				'sSendText' => Phpfox::getPhrase('megapoke.poke_send', array('user_name' => $aUser['full_name'])),
				'pokeFormUrl' => Phpfox::getLib('url')->makeUrl('megapoke')
			)
		);
		
		
			return 'block';	
 
 
 
	}
	
	
}


?>

upload/module/megapoke/template/default/controller/admincp/index.html.php
<?php 
 /**
 * 
 * @copyright		phpfoxLABS.com
 * @author  		Jacob Adams
 * @package  		MEGAPOKE
 * @version 		v1.0.0 - 2011-04-14
 *
 * This work is licensed under a Creative Commons Attribution-ShareAlike 3.0 Unported License.
 * human-readable summary: http://creativecommons.org/licenses/by-sa/3.0/
 * full legal code: http://creativecommons.org/licenses/by-sa/3.0/legalcode
 *
 */
defined('PHPFOX') or exit('NO DICE!'); 

?>

	<form method="post" action="{url link='admincp.megapoke'}cmd_add">
		<div class="table_header">
			phpfoxLABS MegaPoke Types
		</div>

		
        <table width="100%" border="0" cellspacing="0" cellpadding="0">
          <tr>
            <td>Add Poke Type:</td>
            <td><input name="poke_text" type="text" size="50" value=""  />  <input type="submit" value="Add" class="button" /></td>
          </tr>
        
        {if !count($aPokes)} {else}
        {foreach from=$aPokes item=v} 
        
            <tr>
                <td></td>
                <td>{$v.poke_text} | <a href="{url link='admincp.megapoke'}cmd_delete/id_{$v.poke_type_id}">Delete</a></td>
            </tr>
              
          
        {/foreach}
        {/if}
        </table>

	</form>


upload/module/megapoke/include/component/controller/index.class.php
<?php
 /**
 * 
 * @copyright		phpfoxLABS.com
 * @author  		Jacob Adams
 * @package  		MEGAPOKE
 * @version 		v1.0.0 - 2011-04-14
 *
 * This work is licensed under a Creative Commons Attribution-ShareAlike 3.0 Unported License.
 * human-readable summary: http://creativecommons.org/licenses/by-sa/3.0/
 * full legal code: http://creativecommons.org/licenses/by-sa/3.0/legalcode
 *
 */

defined('PHPFOX') or exit('NO DICE!');


#error_reporting(E_ERROR | E_WARNING | E_PARSE);

class Megapoke_Component_Controller_Index extends Phpfox_Component
{
    public function process()
    {        
	
	
		
		if ($_POST['poke_user'] and $_POST['poke_userid'] and $_POST['poke_type_id'])
		{
			
			
			# Check User Privacy Settings
			if (!Phpfox::getService('user.privacy')->hasAccess($_POST['poke_userid'], 'megapoke.who_can_poke'))
			{
				$this->url()->send($_POST['poke_user'], null, Phpfox::getPhrase('megapoke.user_turned_off_megapoke'));
				exit;
			}		
		
			
			
			if (Phpfox::getUserId() == $_POST['poke_userid'])
			{
			
				$this->url()->send($_POST['poke_user'], null, Phpfox::getPhrase('megapoke.you_can_not_send_yourself_a_megapoke'));
				exit;
			}
			
			
			
			# Flood Control: 2 mins
			$time = time() - 120;
			
			$aFlood =phpfox::getLib('phpfox.database')->select('*')
			->from(Phpfox::getT('feed'))
			->where("type_id = 'megapoke' and user_id = '".Phpfox::getUserId()."' and  time_stamp > '".$time."' ")
			->execute('getSlaveRow');	
			if (isset($aFlood['feed_id']))
			{
				
				$this->url()->send($_POST['poke_user'], null, Phpfox::getPhrase('megapoke.too_many_megapokes'));
				exit;
				
			}
				
				
				
			# Get Poke Type
			$aPoke =phpfox::getLib('phpfox.database')->select('*')
			->from(Phpfox::getT('megapoke'))
			->where("poke_type_id = '".Phpfox::getLib('phpfox.database')->escape($_POST['poke_type_id'])."' ")
			->execute('getSlaveRow');	
			
			$d = array();
			$d['content'] = $aPoke['poke_text'];
			
			
			$oUrl = Phpfox::getLib('url');
			$oParseOutput = Phpfox::getLib('parse.output');	
			
			
			#Insert into Feed
			if (Phpfox::getUserParam('megapoke.showpokesinfeed'))
			{ 
				
				
				
				$d['type_id'] = 'megapoke';
				$d['user_id'] = Phpfox::getUserId();
				$d['item_user_id'] = Phpfox::getLib('phpfox.database')->escape($_POST['poke_userid']);
				$d['time_stamp'] = time();
				
	
				phpfox::getLib('phpfox.database')->insert(Phpfox::getT('feed'), $d);
		
		
						
				if(Phpfox::isModule('modsfox'))
				{
					
						$ryansFeedStatus = Phpfox::getPhrase('megapoke.ryans_feed_status', array(
						'viewer_link' => $oUrl->makeUrl('')."". $_POST['poke_user'],
						'viewer_full_name' => Phpfox::getLib('parse.output')->clean($_POST['poke_user_full_name']),
						'content' => $aPoke['poke_text'])
						);	
								
						$sType = 'status';
						$js_user = Phpfox::getUserId();
						$js_mood = "";
						Phpfox::getService('modsfox.status')->add($sType,$ryansFeedStatus,$js_mood,$js_user);
						
				}
							
			
				
				
				
				
			}
			
			
			# Add Notification
			if (Phpfox::isModule('notification'))
			{
				Phpfox::getService('notification.process')->add('megapoke', $aPoke['poke_type_id'], $_POST['poke_userid'], Phpfox::getUserId());
			}
			
			
			
			# Get User Info
			$aUserFrom = Phpfox::getLib('phpfox.database')->select(str_replace("u.", "", Phpfox::getUserField()))
			 ->from(Phpfox::getT('user'))
			 ->where("user_id = '". Phpfox::getUserId()."'")
			 ->execute('getSlaveRow');
			 
			 $d['user_link'] =  $oUrl->makeUrl('')."". $aUserFrom['user_name'];
			 $d['owner_full_name'] = $aUserFrom['full_name'];
			 
	
			# Send Email 
			Phpfox::getLib('mail')->to($_POST['poke_userid'])
			->subject(array('megapoke.newpoke_subject', $d))
			->message(array('megapoke.newpoke_message', $d))
			->notification('megapoke.new_poke_email')
			->send();	
			
			
			
			# Redirect User
			$this->url()->send($_POST['poke_user'], null, Phpfox::getPhrase('megapoke.poke_sent'));
		
		
			
			exit;
			
			
			
			
			
			
		}
		
		
		
		
		exit;
		

    }
	
	
	
	
	
	
}

?> 